cljfmt/src/cljfmt/tool.clj

```diff
@@ -136,18 +180,45 @@
 (defn- recursively-find-files [{:keys [paths] :as options}]
   (mapcat #(find-files options %) (set paths)))
 
+(defmethod report-cli :fix/initial [_ context _] context)
+
+(defmethod report-cli :fix/file [_ context data]
+  (print-file-status context data)
+  (update context :results conj data))
+
+(defmethod report-cli :fix/summary [_ summary _] summary)
+
+(defmethod report-lib :fix/initial [_ context _] context)
+
+(defmethod report-lib :fix/file [_ context data]
+  (update context :results conj data))
+
+(defmethod report-lib :fix/summary [_ summary _] summary)
+
 (defn fix-no-config
   "The same as `fix`, but ignores dotfile configuration."
-  [options]
-  (let [files (recursively-find-files options)
-        map*  (if (:parallel? options) pmap map)]
-    (->> files
-         (map* (partial fix-one options))
-         (map (partial print-file-status options))
-         dorun)))
+  [{:keys [parallel? report] :as options
+    :or   {report report-cli}}]
+  (let [context (report :fix/initial options nil)
+        files   (recursively-find-files context)
+        map*    (if parallel? pmap map)
+        summary (->> files
+                     (map* (partial fix-one context))
+                     (reduce #(report :fix/file %1 %2) context)
+                     doall)]
```

Comment:
`doall` isn't required if there is a `reduce` before it.

Reply:
Ah, yep. Meant to revisit that. Thanks!